pkg/common/cliprinter/cliprinter.go

```diff
@@ -0,0 +1,148 @@
+package cliprinter
+
+import (
+	"io"
+	"os"
+
+	"github.com/spiffe/spire/pkg/common/cliprinter/errorjson"
+	"github.com/spiffe/spire/pkg/common/cliprinter/errorpretty"
+	"github.com/spiffe/spire/pkg/common/cliprinter/protojson"
+	"github.com/spiffe/spire/pkg/common/cliprinter/protopretty"
+	"github.com/spiffe/spire/pkg/common/cliprinter/structjson"
+	"github.com/spiffe/spire/pkg/common/cliprinter/structpretty"
+	"google.golang.org/protobuf/proto"
+)
+
+type Printer interface {
+	MustPrintError(error)
+	MustPrintProto(...proto.Message)
+	MustPrintStruct(...interface{})
+
+	PrintError(error)
+	PrintProto(...proto.Message)
+	PrintStruct(...interface{})
+}
+
+type CustomPrettyFunc func(...interface{})
+
+type printer struct {
+	format formatType
+
+	stdout io.Writer
+	stderr io.Writer
+
+	cp CustomPrettyFunc
+}
+
+func newPrinter(f formatType) *printer {
+	return &printer{
+		format: f,
+		stdout: os.Stdout,
+		stderr: os.Stderr,
```

Comment:
The test suite was just manipulating these directly. I changed things around a little bit, maybe it looks better to you now? If not, lmk what you'd like to see here

Reply:
> These seem like something we'd want to inject for unit-testability?

I just realized maybe you're talking about injecting them from another package, e.g. testing the output of a CLI command in the tests for that command. If that's the case, I think the better use is the interface? We probably don't want CLI tests to have to know about how the printer is formatting or displaying things... just that the printer works (because it is tested) and that we called it?

Reply:
Yeah, that makes sense. Going this route will require some finagling of existing tests which assert the output but that's ok.